### ISSUE TITLE
Incorrect parsing of image source that uses a registered custom scheme

### ISSUE THREAD

**Issue by gfox1984** [mganss/HtmlSanitizer#118]
When I register the custom scheme called "resources":
`sanitizer.AllowedSchemes.Add("resources");`

then sanitizes this string: 
`<IMG src="resources://ais_w20_h20_33ba129c.png">`

I get the following output:
`<img src="resources://ais_w20_h20_33ba129c.png/">`

Notice the unwanted slash after the image name.
This is a problem for me because the "resources" scheme is replaced later with an actual URL, and that final slash is breaking the resulting URL...

You can see the fiddle here: https://dotnetfiddle.net/UPzTkf

**Comment by mganss** [comment]
This occurs because we are using `System.Uri` to parse and render URLs. I've already pushed a fix for this as well as #106 and #66.

**Comment by gfox1984** [comment]
Cool. Do you have any timeframe for a release and is there any workaround I can use in the meantime?

**Comment by gfox1984** [comment]
OK I found a workaround inspired from #106:

```
var sanitizer = new HtmlSanitizer();
//sanitizer.AllowedSchemes.Add("resources");
sanitizer.RemovingAttribute += (s, e) =>
    	e.Cancel = e.Reason == RemoveReason.NotAllowedUrlValue
    	&& e.Attribute.Value.StartsWith("resources://"); 
```

Fiddler here: https://dotnetfiddle.net/gZxP1I

**Comment by mganss** [comment]
3.5.168-beta includes the fix.

**Comment by gfox1984** [comment]
Hi, do you have an ETA for the release of a stable 3.5?

**Comment by mganss** [comment]
There are incompatible API changes from 3.4, so per [SemVer](http://semver.org/) this would mean we'd have to increase the major version. I had originally planned to update to 4.0 once AngleSharp 0.10 was released but AngleSharp currently seems to be in a semi-abandoned state (first CsQuery was abandoned and now this 🙄 ). I guess I'll just go ahead and release the current state of 3.5 as 4.0 and if AngleSharp 0.10 ever gets released we'll increase the major version again. Objections?

**Comment by gfox1984** [comment]
No objections here:) !

**Comment by mganss** [comment]
Released 4.0.179
